Added documentation to some `add_*_plugin` methods

RoniPerson opened the pull request:
Added documentation to some `add_*_plugin` methods and the corresponding `remove_*_plugin` methods.
Added an example of how to register a plugin to `add_inspector_plugin` and linked to it in the other methods.
An c# example is missing because I don't use mono versions so I couldn't test it.

RoniPerson commented:
Can someone tell me what is the problem with the formating?

pycbouh commented:
> Can someone tell me what is the problem with the formating? I am not an expert with the format but it looks fine to me.

It's telling you that an empty line mustn't have any indentation.

RoniPerson commented:
Ok i will fix this

RoniPerson commented:
> Also, please squash your commits. Your PR must only contain one commit.

I will do this when the format check passed

RoniPerson commented:
f

RoniPerson committed:
Added documentation to some `add_*_plugin` methods

Added documentation to some `add_*_plugin` methods and the corresponding `remove_*_plugin` methods.
Added an example of how to register a plugin to `add_inspector_plugin` and linked to it in the other methods.

akien-mga reviewed (approved):
Sorry for the delay. This looks great :)

akien-mga commented:
Thanks!

pycbouh commented on doc/classes/EditorPlugin.xml:
```diff
@@ -100,6 +101,9 @@
 			<argument index="0" name="importer" type="EditorImportPlugin">
 			</argument>
 			<description>
+				Registers a new import plugin. Import plugins are used to add a new [Resource] which can be imported. See [EditorImportPlugin] for more information.
```
I think all the "See `ClassName` for more information" can be removed and instead the opening sentence can be formatted as "Register a new `ClassName`". This includes the existing description for `add_export_plugin` which you used as an example for your new descriptions.

```suggestion
				Registers a new [EditorImportPlugin]. Import plugins are used to import custom and unsupported assets as a custom [Resource] type.
```

pycbouh commented on doc/classes/EditorPlugin.xml:
```diff
@@ -100,6 +101,9 @@
 			</argument>
 			<description>
+				Registers a new import plugin. Import plugins are used to add a new [Resource] which can be imported. See [EditorImportPlugin] for more information.
+				[b]Note:[/b] If you want to import custom 3d files have a look at [method add_scene_import_plugin] instead.
```
```suggestion
				[b]Note:[/b] If you want to import custom 3D asset formats use [method add_scene_import_plugin] instead.
```

pycbouh commented on doc/classes/EditorPlugin.xml:
```diff
@@ -108,6 +112,20 @@
 			<argument index="0" name="plugin" type="EditorInspectorPlugin">
 			</argument>
 			<description>
+				Registers a new inspector plugin. Inspector plugins are used to extend the default inspector. See [EditorInspectorPlugin] for more information.
```
```suggestion
				Registers a new [EditorInspectorPlugin]. Inspector plugins are used to extend [EditorInspector] and provide custom configuration tools for your object's properties.
```

pycbouh commented on doc/classes/EditorPlugin.xml:
```diff
@@ -92,6 +92,7 @@
 			</argument>
 			<description>
 				Registers a new export plugin. Export plugins are used when the project is being exported. See [EditorExportPlugin] for more information.
```
```suggestion
				Registers a new [EditorExportPlugin]. Export plugins are used to perform tasks when the project is being exported.
```

pycbouh commented on doc/classes/EditorPlugin.xml:
```diff
@@ -108,6 +112,20 @@
 			</argument>
 			<description>
+				Registers a new inspector plugin. Inspector plugins are used to extend the default inspector. See [EditorInspectorPlugin] for more information.
+				[b]Note:[/b] Always use [method remove_inspector_plugin] to remove the [code]EditorInspectorPlugin[/code] if the [code]EditorPlugin[/code] is disabled.
```
Weird use of `[code]` tags. Just use links.

```suggestion
				[b]Note:[/b] Always use [method remove_inspector_plugin] to remove the registered [EditorInspectorPlugin] when your [EditorPlugin] is disabled to prevent leaks and an unexpected behavior.
```

pycbouh commented on doc/classes/EditorPlugin.xml:
```diff
@@ -116,6 +134,7 @@
 			<argument index="0" name="scene_importer" type="EditorSceneImporter">
 			</argument>
 			<description>
+				Registers a new scene importer. Scene importers can import custom 3d formats as scenes. See [EditorImportPlugin] for more information.
```
```suggestion
				Registers a new [EditorSceneImporter]. Scene importers are used to import custom 3D asset formats as scenes.
```

pycbouh commented on doc/classes/EditorPlugin.xml:
```diff
@@ -124,6 +143,8 @@
 			<argument index="0" name="plugin" type="EditorNode3DGizmoPlugin">
 			</argument>
 			<description>
+				Registers a new gizmo plugin. Gizmo plugins are used to add custom gizmos to a [Node3D]. See [EditorNode3DGizmoPlugin] for more information.
```
```suggestion
				Registers a new [EditorNode3DGizmoPlugin]. Gizmo plugins are used to add custom gizmos to the 3D preview viewport for a [Node3D].
```

YuriSizov reviewed (changes requested):
Left some suggestions, good to merge after those are addressed.

pycbouh commented:
Ah damn, @akien-mga :(

akien-mga commented:
Oops :D follow-up PR it will be :)

akien-mga commented:
Cherry-picked for 3.4.

akien-mga commented:
Cherry-picked for 3.3.3.
